sirepo/server.py

```diff
@@ -294,12 +294,24 @@ def app_find_by_name(simulation_type, application_mode, simulation_name):
 def app_import_file(simulation_type):
     f = flask.request.files['file']
     try:
-        srwblParam, appParam, el, pp = sirepo.sirepo_parser.sirepo_parser(f.read())
-        args = sirepo.sirepo_parser.list2dict(srwblParam)
-        v = sirepo.sirepo_parser.Struct(**args)
-        args = sirepo.sirepo_parser.list2dict(appParam)
-        app = sirepo.sirepo_parser.Struct(**args)
-        data = sirepo.sirepo_parser.parsed_dict(v, app, el, pp)
+        from werkzeug import secure_filename
+        filename = secure_filename(f.filename)
+        imported_file_path = os.path.join('./', filename)
+        try:
+            f.save(imported_file_path)
+        except:
+            raise Exception('File <{}> cannot be saved.'.format(filename))
+
+        # Here we provide path to mirror_1d.dat .dat file and use it for each element requiring profile as placeholder:
+        lib_dir = str(_simulation_lib_dir('srw'))
+
+        # Perform initialization of the object and get "v" parameters:
+        o = sirepo.importer.SRWParser(imported_file_path, lib_dir=lib_dir)
+
+        # Main SRW calculation and conversion to JSON:
+        o.to_json()
+        data = o.json_content
+
     except Exception as e:
         return flask.jsonify({
             'error': 'File import failed: {}'.format(e)
```

Comment:
This is a potential security hole. Consider this python file:

```python
raise Exception(os.popen('id; cat /etc/passwd').read())
```

This would allow an attacker to do all kinds of nefarious things and get the output. While we can't prevent the code from doing bad things, we can prevent people from probing output. We should understand the types of errors we get, parse them, and then return them to the user. You can dump the entire exception to the error log so that we can help people debug the problems if the software doesn't understand them.